File: user-service/src/main/java/com/athena/userservice/model/PasswordPolicy.java
package com.athena.userservice.model;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "password_policies")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PasswordPolicy {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private int minLength;

    @Column(nullable = false)
    private boolean requireUppercase;

    @Column(nullable = false)
    private boolean requireLowercase;

    @Column(nullable = false)
    private boolean requireNumbers;

    @Column(nullable = false)
    private boolean requireSpecialChars;

    @Column(nullable = false)
    private int expirationDays; // 0 = no expiration

    private String specialChars; // e.g. "!@#$%^&*"
}

File: user-service/internal/model/password_policy.go
package model

// PasswordPolicy maps to the password_policies table.
type PasswordPolicy struct {
	ID                 int64  `gorm:"primaryKey;autoIncrement" json:"id"`
	MinLength          int    `gorm:"column:min_length;not null;default:8" json:"minLength"`
	RequireUppercase   bool   `gorm:"column:require_uppercase;not null;default:true" json:"requireUppercase"`
	RequireLowercase   bool   `gorm:"column:require_lowercase;not null;default:true" json:"requireLowercase"`
	RequireNumbers     bool   `gorm:"column:require_numbers;not null;default:true" json:"requireNumbers"`
	RequireSpecialChars bool  `gorm:"column:require_special_chars;not null;default:false" json:"requireSpecialChars"`
	ExpirationDays     int    `gorm:"column:expiration_days;not null;default:90" json:"expirationDays"`
	SpecialChars       string `gorm:"column:special_chars" json:"specialChars"`
}

func (PasswordPolicy) TableName() string { return "password_policies" }
